### DISCUSSION TITLE
how can I log true position messages as csv file?

### DISCUSSION THREAD

**Discussion by lwisle** [USC-ACTLab/crazyswarm#597] [Q&A] [answered]
Hello!
Can crazyswarm record position messages from VRPN as csv file? 
If I modify **write_csvs: False** to **write_csvs: True**,what will be happen? Which message will be recorded?

what's more, if I use optitrack + vrpn + crazyswarm, is the parameter locSrv.x in picture same with the true position x in vrpn or optitrack?
Thanks for your help.
![image](https://user-images.githubusercontent.com/82260604/154834748-53a61bde-6d48-4cec-b2a7-01596c03a96c.png)

**Comment by whoenig** [comment, marked as answer]
The `write_csvs` option is legacy and should not be used. If you want to record data, I'd record the `/tf` messages using `rosbag`.

You are correct about the logging parameters. Note that they'll be delayed from the `/tf` messages, as they first need to be communicated to the Crazyflie (and back to the PC if you log them). So these variables are mostly interesting, if you need to compare the values in the same time domain (e.g., if you work on the EKF state estimator).

**Reply by lwisle** [reply]
Thank you!
